contracts/0.4.24/utils/Versioned.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-FileCopyrightText: 2023 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.4.24;
+
+import "@aragon/os/contracts/common/UnstructuredStorage.sol";
+
+
+contract Versioned {
+    using UnstructuredStorage for bytes32;
+
+    event ContractVersionSet(uint256 version);
+
+    /// @dev Storage slot: uint256 version
+    /// Version of the initialized contract storage.
+    /// The version stored in CONTRACT_VERSION_POSITION equals to:
+    /// - 0 right after the deployment, before an initializer is invoked (and only at that moment);
+    /// - N after calling initialize(), where N is the initially deployed contract version;
+    /// - N after upgrading contract by calling finalizeUpgrade_vN().
+    bytes32 internal constant CONTRACT_VERSION_POSITION = keccak256("lido.Versioned.contractVersion");
+
+    uint256 internal constant PERTIFIED_VERSION_MARK = uint256(-1);
+
+    constructor() {
+        // lock version in the implementation's storage to prevent initialization
+        CONTRACT_VERSION_POSITION.setStorageUint256(PERTIFIED_VERSION_MARK);
+    }
+
+    /// @notice Returns the current contract version.
+    function getContractVersion() public view returns (uint256) {
+        return CONTRACT_VERSION_POSITION.getStorageUint256();
+    }
+
+    function _checkContractVersion(uint256 version) internal view {
+        require(version == getContractVersion(), "UNEXPECTED_CONTRACT_VERSION");
+    }
+
+    function _writeContractVersion(uint256 version) internal {
```

Comment:
Let's add a comment that logic related to `InvalidContractVersionIncrement` is not added here to save contract size for 0.4.24 overweighed contracts.